packages/coil-extension/src/content/services/ContentScript.ts

```diff
@@ -134,6 +135,11 @@ export class ContentScript {
         } else if (request.command === 'setMonetizationState') {
           this.monetization.setState(request.data)
         } else if (request.command === 'monetizationProgress') {
+          if (this.monetization.getState() === 'pending') {
+            // Don't transition from stopped → started. This prevents catch-up
```

Comment:
@sentientwaffle 
What's happening here ?
At a shallow reading, it seems implied that this supports progress events which may potentially happen when the state is stopped (i.e. tab isn't active) ?

Reply:
If you're deactivating a tab, should it not just pause everything ? It seems weird that it would suddenly "catchup" pay as soon as you switch the tab, and that the events should be emitted when the user is not watching (tab is not active). Especially when you consider it would otherwise wait?

Am I misunderstanding something

Reply:
> At a shallow reading, it seems implied that this supports progress events which may potentially happen when the state is stopped (i.e. tab isn't active) ?

Yes, that may happen in order to prevent incorrectly activating a deactivated tab when a late payment arrived. And I figured emitting the progress event (even when state=stopped) was better than just discarding it (and much simpler than queuing them up until the resume).

> If you're deactivating a tab, should it not just pause everything ? It seems weird that it would suddenly "catchup" pay as soon as you switch the tab, and that the events should be emitted when the user is not watching (tab is not active). Especially when you consider it would otherwise wait?

Yeah, I'll fix that.